includes/gateway/class-omise-payment-billpayment-tesco.php

```diff
@@ -137,84 +137,92 @@ public function display_barcode( $order, $context = 'view' ) {
-				</small>
-
+				<!-- Display for printing only. -->
+				<!-- Hidden by assets/css/omise-billpayment-print.css. -->
 				<?php if ( 'email' !== $context ) : ?>
-					<div class="omise-billpayment-tesco-print-button">
-						<button onClick="window.print()" class="button button-primary">Print barcode</button>
-					</div>
-				<?php endif; ?>
-			</div>
-
-			<!-- Display for printing only. -->
-			<!-- Hidden by assets/css/omise-billpayment-print.css. -->
-			<?php if ( 'email' !== $context ) : ?>
-
-				<div class="omise-billpayment-tesco-print-detail">
-					<div>
-						<h2><?php echo apply_filters( 'woocommerce_thankyou_order_received_text', __( 'Thank you. Your order has been received.', 'woocommerce' ), $order ); ?></h2>
-					</div>
-
 					<div class="omise-billpayment-tesco-print-order-overview">
-						<div class="omise-billpayment-tesco-print-order-overview-item">
+						<div>
 							<?php _e( 'Order number:', 'woocommerce' ); ?>
 							<br/><strong><?php echo $this->order()->get_order_number(); ?></strong>
 						</div>
 
-						<div class="omise-billpayment-tesco-print-order-overview-item">
-							<?php _e( 'Date:', 'woocommerce' ); ?>:
+						<div>
+							<?php _e( 'Date:', 'woocommerce' ); ?>
 							<br/><strong><?php echo wc_format_datetime( $this->order()->get_date_created() ); ?></strong>
 						</div>
 
-						<?php if ( is_user_logged_in() && $this->order()->get_user_id() === get_current_user_id() && $this->order()->get_billing_email() ) : ?>
-							<div class="omise-billpayment-tesco-print-order-overview-item">
-								<?php _e( 'Email:', 'woocommerce' ); ?>:
-								<br/><strong><?php echo $this->order()->get_billing_email(); ?></strong>
-							</div>
-						<?php endif; ?>
-
-						<div class="omise-billpayment-tesco-print-order-overview-item">
-							<?php _e( 'Total:', 'woocommerce' ); ?>:
+						<div>
+							<?php _e( 'Total:', 'woocommerce' ); ?>
 							<br/><strong><?php echo $this->order()->get_formatted_order_total(); ?></strong>
 						</div>
-					</div>
 
-					<?php if ( $this->order()->get_payment_method_title() ) : ?>
-						<div class="omise-billpayment-tesco-print-order-payment-method">
-							<?php _e( 'Payment method:', 'woocommerce' ); ?>
-							<br/><strong><?php echo wp_kses_post( $this->order()->get_payment_method_title() ); ?></strong>
-						</div>
-					<?php endif; ?>
+						<?php if ( $this->order()->get_payment_method_title() ) : ?>
+							<div>
+								<?php _e( 'Payment method:', 'woocommerce' ); ?>
+								<br/><strong><?php echo wp_kses_post( $this->order()->get_payment_method_title() ); ?></strong>
+							</div>
+						<?php endif; ?>
+					</div>
+				<?php endif; ?>
+				<!-- /END .omise-billpayment-tesco-print-detail -->
+
+				<p>
+					<?php
+					echo sprintf(
+						wp_kses(
+							__( 'Please bring this barcode to pay at Tesco Lotus by:<br/><strong>%1$s %2$s</strong>.', 'omise' ),
+							array( 'br' => array(), 'strong' => array() )
```

Comment:
Do WooCommerce stores have a set DATETIME format in their settings?

Reply:
@jonrandy yup (in fact, WordPress does)
👇 from WordPress General Setting page.
![Screen Shot 2562-08-15 at 12 32 01](https://user-images.githubusercontent.com/2154669/63074716-b3700100-bf58-11e9-9989-688a275c58da.png)

Reply:
Just to give you a bit more context.
The following lines
```php
wc_format_datetime( $expires_datetime, wc_date_format() );
wc_format_datetime( $expires_datetime, wc_time_format() );
```

The `$expires_datetime` variable comes from
```php
$expires_datetime   = new WC_DateTime( $charge['source']['references']['expires_at'], new DateTimeZone( 'UTC' ) );
```

And, `WC_DateTime` extends from a built-in PHP `DateTime` class.

So basically we pass Omise's `expires_at` (as ISO 8601 Zulu time format) to `WC_DateTime` class (that extended from `DateTime`) and set its default TZ to `UTC`.

Then, we use `wc_date_format()` and `wc_time_format()` to retrieve WordPress Date and Time formatting from its configuration and pass those to `wc_format_datetime()` function, which does a DateTime formatting job.

Reply:
So, given the above image, it would appear that Wordpress has a date format, and a time format, but no specific datetime format

Reply:
Forgot to mention, WordPress also provides TimeZone configuration, which we retrieve and config to `WC_DateTime` at the earlier part of code.
```php
$expires_datetime = new WC_DateTime( $charge['source']['references']['expires_at'], new DateTimeZone( 'UTC' ) );
$expires_datetime->set_utc_offset( wc_timezone_offset() );
```
![Screen Shot 2562-08-15 at 12 41 50](https://user-images.githubusercontent.com/2154669/63074992-1f9f3480-bf5a-11e9-90dc-8f2def44fa61.png)

Reply:
@jonrandy ok I'm not followed now 😂 
--edited
Got it. You were looking at `<strong>%1$s %2$s</strong>` right?

Reply:
Yes. If there was a DATETIME format available, I would expect a single string here